### ISSUE TITLE
Slow

### ISSUE THREAD

**Issue by campones** [webmin/webmin#1069]
I m sorry to tell you this, but since the last update 1.910, for me, webmin became very slow loading many pages.  happens on different servers

not sure if anyone noticed this?

**Comment by swelljoe** [comment]
Seems fine, to me, on all of our servers.

You'll need to be more specific about what page(s) you see this problem on. Also would be useful to know if CPU usage spikes on the server when the pages are loading (check `top`), or if the browser is working hard.

What browser? Can you try a different browser to see if there's a difference?

And, while it's probably not relevant, include the OS you're running Webmin on.

**Comment by cagivacode** [comment]
I have noticed this as well (I thought it was just me) - a fresh install of 1.910 was like that so I skipped updating some other panels....

I can be more specific.

In my case it happens on small VMs (640k / 1 core / 8gig) -- minimal install -- UBUNTU 18.04.

Most noticeable getting into the panel (after login page) and I have it set at default -- so it goes to the dashboard.

Top shows 7-8 /usr/share/webm*.cgi processes all at 30% cpu.......page is partially rendered (left side bar is there with menu / main panel is blank / red progress bar is going back and forth). It can stay CPU bound for minutes.

Exact same spec VMs on same host, but with slackware or CENTOS do not seem to have a problem.

Since this is on a internal network....network speed is not a factor. Have tried Chrome / Chromium based / MS EDGE / MSIE....also not a factor.
 
The VM gets so bound up it is hard to tell....but I think it is fighting apt-get checking for software updates (just my impression).

webmin1.910   Authentic theme version 19.34

**Comment by campones** [comment]
for example, when I entered the main page, I keep seeing (waiting for [ip of server:10000]) or when I open a cronjob , takes age with no apparent reason (server is not under heavy load). 

Also I have sometimes issues with loading the java manager, I see http timeout stuff, all this I never testified before

**Comment by campones** [comment]
exactly as cagivacode said "partially rendered page" I see this in the cronjob

**Comment by rostovtsev** [comment]
> Exactly as cagivacode said "partially rendered page" I see this in the cronjob

Is this happening to you only in Cron Jobs module? There were fixes that Jamie did recently to address this issue, for some configurations.

@campones @cagivacode Does it happen to you guys, only on loading Dashboard and Cron Jobs module or any other module too? For example, if you go to _Systems/Change Passwords_ does it load slow too?

**Comment by cagivacode** [comment]
Well you certainly have 2 extreme points of view (my small VMs and campones baremetal with grey theme).

For me it is mainly first login with 7-8 webmin processes (most are index.cgi) all at 30% cpu.

After that switching modules (chron / passwords) takes a bit (red progress bar almost makes it across the page).........but nothing like the login dash which sits for an extended period.

going to software package updates (UBUNTU) can take five minutes -- based on the server's other functions and loads. That is apt-show-versions siting at 75 - 100% cpu....for minutes.

I did a little testing and I find sitting in a module (e.g. chron ) I see in top --- command shows [/usr/share/webm] <defunct> and it periodically hits 45% cpu.

Switching into chron sometimes shows several 404.cgi processes as well.

**Comment by campones** [comment]
it didn't happen right now when I opened the password change page, but I waited 20 good second to load the dashboard, as cagivacode said, left menu open but we get a blank main page

**Comment by campones** [comment]
when I open a cronjob , I also see a partial rendering like this 
[https://i.gyazo.com/596118723e6af3ca57b25911bc64e300.png](url)

then need to wait for the rest to appear

**Comment by rostovtsev** [comment]
What if you disable Disk Usage tB for Dashboard in _Disk and Network Filesystems_ (_mount_) module's configuration?

**Comment by rostovtsev** [comment]
> When I open a cronjob , I also see a partial rendering like this..

This bug should be fixed.
https://github.com/authentic-theme/authentic-theme/issues/1339

**Comment by campones** [comment]
> > When I open a cronjob , I also see a partial rendering like this..
> 
> This bug should be fixed.
> [authentic-theme/authentic-theme#1339](https://github.com/authentic-theme/authentic-theme/issues/1339)

I m not using authentic theme

**Comment by rostovtsev** [comment]
Still, it was incorrectly submitted. It's Webmin bug.

**Comment by campones** [comment]
> What if you disable _Disk Usage_ tab for Dashboard in _Disk and Network Filesystems_ (_mount_) module's configuration?

I don't know how to do that sorry

**Comment by campones** [comment]
https://i.gyazo.com/e27f3b1158da34aad84a806fe4b935ba.png

is it this?

**Comment by rostovtsev** [comment]
Show available disk data. Set it to No.

Is there any reason why you're using Gray Theme still?

**Comment by campones** [comment]
it's better?

**Comment by campones** [comment]
i do not need the fancy stuff, just clear view and fast loading  is what matters to me

**Comment by rostovtsev** [comment]
Yes, recent versions of Authentic Theme same fast or faster.

Did changing that option do anything?

**Comment by campones** [comment]
no it didn't.  I ll narrow down the problem to the cron job page for example. I still have some servers with 1900 installed and the difference is obvious when I open a cronjob, it loads instantly, doesn't really make difference for the dashboard however

**Comment by cagivacode** [comment]
I just noticed that this board did not show my earlier comment correctly.....

` did a little testing and I find sitting in a module (e.g. chron ) I see in top --- command shows [/usr/share/webm] <defunct> and it periodically hits 45% cpu.`

Switching to chron page (while monitoring by ssh with top open and in command mode) there is orphaned "defunct" and then zombie processes using cpu cycles. Even if you don't click on anything.

I think switching "pages" is leaving multiple stats.cgi processes beheaded.

rdl

**Comment by campones** [comment]
well I m back on 1900 release and all is fine, thanks!

**Comment by mattyclark1981** [comment]
Something I have a problem with too. Three servers, one on the internet (Ubuntu 16.04), two on local network only (Ubuntu 18.04). All are currently running Webmin 1.910. Same in Chrome or Edge. 

It's not limited to particular modules, or particular versions, but could be theme related, as they all run "Authentic". 

I've seen it to some extent on pretty much every release in the last 18 months or so. 

The Dashboard can take up to 10+ seconds to display. Sometimes its only 3 or 4.

I've disabled IPv6 lookups in APT on two of the three (one on the internet, one local), and that seemed to help a little, but still too early to tell.

**Comment by rostovtsev** [comment]
@mattyclark1981 You could check if it's an issue by switching to Gray Theme. Although, it doesn't look at all as theme related issue.

> It's not limited to particular modules, or particular versions, but could be theme related, as they all run "Authentic".

What exactly doesn't work for you, what worked before 1.910 release? Authentic Theme as fast and stable as ever, I don't experience any issues with it, while running it on multiple servers.

**Comment by mattyclark1981** [comment]
Everything works, but as per the thread subject, it's slow. I can remember a while back (although not an exact version number, sadly) where everything was pretty snappy. I'd say less than a second to load the dashboard on the two servers on the local network. 

But it seems to have gotten slower and slower to load.

I'll try the grey theme.

**Comment by campones** [comment]
I think your issue is different, I specifically said that the problem came with 1910

**Comment by rostovtsev** [comment]
Okay, what if you try to go to _Webmin/Webmin Configuration/Web Server Options_. 

What is _Gzip compress static files?_ set to? Try to set to _Never_ if first or last options don't work.

**Comment by mattyclark1981** [comment]
OK... Initial results are promising. All three servers were set to "_Only if pre-compressed .gz file exists_". 

I've changed that to "_Never_", and it seems to have helped a lot. Back in the <1s load time now. I'll gather some more data over the coming days, and update.

Thank you Rostovtsev!

**Comment by rostovtsev** [comment]
You are welcome.

However, it's unclear to me why exactly this happening.

What if you set it to _Dynamic_ mode?

**Comment by gnadelwartz** [comment]
i have the same problem on suse, Apache Reverse Proxy. imho the problem appeared after a security update if the os. setting gzipped files to never didn't help

**Comment by gnadelwartz** [comment]
you can try it yourself https://mail.rrr.de/webmin/

**Comment by gnadelwartz** [comment]
@iliaross I attached the strace output and a small part of my apache reverse proxy log.
the log contains the strace -ft output for switching to fipv4 firewall. then back to dashboard and then to webmin config.

For some reason the is a long delay between webbrowser requests, see apache-log for the gifs in webmin config.

[webmin.trace.txt](https://github.com/webmin/webmin/files/3573854/webmin.trace.txt)
[apache-log.txt](https://github.com/webmin/webmin/files/3573855/apache-log.txt)

**Comment by gnadelwartz** [comment]
OK, found the problem with help of @iliaross 

I used apache reverse proxy with apache config:

```
   ProxyPass /webmin http://localhost:10000
   ProxyPassReverse /webmin http://localhost:10000
```

but my localhost resolved not to 127.0.0.1. this was an error of my hoster who assigend localhost to a public address which leads to a round robin between the public adress and 127.0.0.1.

The solution was to use 127.0.0.1 instead of localhost in the apache config:

```
   ProxyPass /webmin http://127.0.0.1:10000
   ProxyPassReverse /webmin http://127.0.0.1:10000
```
AND remove the hoster provided default search path in ```/etc/resolv.conf```

I'll also inform the hoster to remove ```localhost.xxxx.de``` from his DNS

**Comment by gnadelwartz** [comment]
@jcameron I propose to change example soache config fir reveree proxyon http://www.webmin.com/apache.html from

ProxyPass /webmin http://localhost:10000
ProxyPassReverse /webmin http://localhost:10000

to

   ProxyPass /webmin http://127.0.0.1:10000
   ProxyPassReverse /webmin http://127.0.0.1:10000

to avoid dns problems like mine for new users

**Comment by rostovtsev** [comment]
Yeah, the same issue may happen for MySQL, so setting _127.0.0.1_ is always seems better to me.
